extension.js

```diff
@@ -110,6 +110,7 @@ function enable() {
     settings.connect('changed::window-stick', set_window_stick);
     settings.connect('changed::window-height', update_window_geometry);
     settings.connect('changed::window-skip-taskbar', set_skip_taskbar);
+    settings.connect('changed::window-maximize', window_maximize);
```

Comment:
For every `connect()` there should be a matching `disconnect()` (or `GObject.signal_handlers_disconnect_by_func()`, see https://github.com/amezin/gnome-shell-extension-ddterm/blob/master/extension.js#L419)